### PR TITLE
Scrape logs from shoot kube-system namespace

### LINKED ISSUE #5439: Scrapping logs from shoot kube-system pods.
**How to categorize this issue?**

/area logging
/kind enhancement

**What would you like to be added**:
It would be good to add the logs from the shoot `kube-system` pods.
As there is already a node `systemd` scraping configuration this could be achieved easily by adding additional job in the  `promtail` [configuration](https://github.com/gardener/gardener/blob/3c6404d23e2fb583a991c0b9247a3f23c9523d93/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go#L107).
1. The easiest way is to use [File Target Discovery](https://grafana.com/docs/loki/latest/clients/promtail/scraping/#file-target-discovery). It is very similar to the [fluent-bit tail plugin](https://github.com/gardener/gardener/blob/3c6404d23e2fb583a991c0b9247a3f23c9523d93/charts/seed-bootstrap/charts/fluent-bit/templates/_fluent-bit-config.tpl#L23) which we already use in the Seed logging stack. We get the same meta information, the `pod_name`, `namespace_name`, `container_name` and the `cointainer_id`. Also, we get the `nodename`. In this forked repo [branch](https://github.com/vlvasilev/gardener/tree/scrape-shoot-kube-system-logs) I have a working version that does not need any extra configuration.
Pros:
- A connection to the kube API server is not required. all of the logs in the `kube-system` are scraped and sent to Loki.
Cons:
- As we have seen a lot of the users are deploying in the `kube-system`. We don't have a convenient way to differentiate them. If start to filter those user logs with regex we have to wait for a new release each time we add a new component in the shoot system.

2. The second way is to use [Kubernetes Discovery](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#kubernetes_sd_config). Here we can get vast metadata like the annotation, the labels and etc.
Pros:
- A lot of metadata so a new release is not required whenever a new component is added.
- The user's logs are not collected.
Cons:
- A connection to the Kube API server is required, which may increase the network traffic cost.

**Why is this needed**:
To speed up issues investigation using the Grafana

### PR DESCRIPTION
**Pull request by vlvasilev**
**How to categorize this PR?**

/area logging
/kind enhancement

**What this PR does / why we need it**:
Logs from pods with labels `origin=gardener` and `gardener.cloud/role` in the shoot `kube-system` namespace are scraped and sent to Loki under the `operator` tenant.

**Which issue(s) this PR fixes**:
Fixes [#5439](https://github.com/gardener/gardener/issues/5439)

**Special notes for your reviewer**:

**Release note**:

```feature operator
Logs of the gardener components in the shoot's `kube-system` are scraped and available for the operators.
```

### CODE REVIEW

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go [hunk 1]
```diff
@@ -137,6 +137,45 @@ var defaultConfig = config{
...
+			"job_name": "containers",
+			"static_configs": []map[string]interface{}{
+				{
+					"targets": []interface{}{
+						"localhost",
+					},
+					"labels": map[string]interface{}{
+						"job":      "kube-system",
+						"nodename": "${HOSTNAME}",
+						"__path__": "/var/log/containers/*_kube-system_*log",
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]:
```suggestion
						"__path__": "/var/log/containers/*_ " + metav1.NamespaceSystem + "_*log",
```

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go [hunk 2]
```diff
@@ -137,6 +137,45 @@ var defaultConfig = config{
...
 		},
+		{
+			"job_name": "containers",
+			"static_configs": []map[string]interface{}{
+				{
+					"targets": []interface{}{
+						"localhost",
+					},
+					"labels": map[string]interface{}{
+						"job":      "kube-system",
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]:
```suggestion
						"job":      metav1.NamespaceSystem,
```

File: hack/local-development/local-garden/kubeconfigs/default-admin.conf [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
     server: https://localhost:2443
+  name: local-garden
```
Comment (rfranzke) [inline hack/local-development/local-garden/kubeconfigs/default-admin.conf]: Why was this file changed?

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/components.go [hunk 4]
```diff
@@ -44,4 +44,5 @@ type Context struct {
 	SSHPublicKeys           []string
 	LokiIngress             string
 	PromtailEnabled         bool
+	ApiServerURL            string
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/components.go]:
```suggestion
	APIServerURL            string
```

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go [hunk 5]
```diff
@@ -48,6 +53,13 @@ func init() {
 	if err != nil {
 		panic(err)
 	}
+	tplPromtail, err = template.
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go:57]:
```suggestion

	tplPromtail, err = template.
```

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go [hunk 6]
```diff
@@ -59,144 +71,21 @@ elif  [ ! -d "$TEMP_JOURNAL_FILE" ] && [ -d "$PERSISTANT_JOURNAL_FILE" ]; then
...
-	conf.Client.Url = "https://" + ctx.LokiIngress + "/loki/api/v1/push"
-	conf.Client.TLSConfig.ServerName = ctx.LokiIngress
-	return conf, nil
-}
-
-func getPromtailConfigurationFile(ctx components.Context) (extensionsv1alpha1.File, error) {
-	conf, err := getPromtailConfiguration(ctx)
-	if err != nil {
-		return extensionsv1alpha1.File{}, err
+		return extensionsv1alpha1.File{}, fmt.Errorf("loki ingress url is missing for %s", ctx.ApiServerURL)
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Shouldn't this be `ctx.LokiIngress` instead of `ctx.ApiServerURL` in the error message?

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go [hunk 7]
```diff
@@ -59,144 +71,21 @@ elif  [ ! -d "$TEMP_JOURNAL_FILE" ] && [ -d "$PERSISTANT_JOURNAL_FILE" ]; then
...
 
-	configYaml, err := yaml.Marshal(&conf)
-	if err != nil {
+	if err := tplPromtail.Execute(&config, map[string]interface{}{
+		"clientURL":     "https://" + ctx.LokiIngress + "/loki/api/v1/push",
+		"pathCACert":    PathCACert,
+		"lokiIngress":   ctx.LokiIngress,
+		"pathAuthToken": PathAuthToken,
+		"apiServerURL":  ctx.ApiServerURL,
+		"apiServerURL2": strings.TrimPrefix(ctx.ApiServerURL, "https://"),
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Can you find a better name for the key?

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go (hunk 1 above)
Comment (vlvasilev) [reply pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Sorry. Forgot to squash the commits.

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go (hunk 6 above)
Comment (vlvasilev) [reply pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: If the `ctx.LokiIngress` is missing then for which cluster the error is for?

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go (hunk 7 above)
Comment (vlvasilev) [reply pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Stupid me, sorry. I forgot to rename it.

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go (hunk 6 above)
Comment (rfranzke) [reply pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Now I got was you try to achieve here. I suggest to just remove this from the error messages, the log message that prints this eventually should have the information about the shoot.

File: hack/local-development/local-garden/kubeconfigs/default-admin.conf (hunk 3 above)
Comment (vlvasilev) [reply hack/local-development/local-garden/kubeconfigs/default-admin.conf]: I don't know, sorry.

File: pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go (hunk 6 above)
Comment (vlvasilev) [reply pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go]: Ok, thanks.

**Review by rfranzke — APPROVED** [review]
/lgtm
